src/v/cluster/controller_api.cc

```diff
@@ -53,6 +53,37 @@ controller_api::get_reconciliation_state(std::vector<model::ntp> ntps) {
     });
 }
 
+ss::future<result<bool>>
+controller_api::all_reconciliations_done(std::vector<model::ntp> ntps) {
+    bool reduced = true;
+    errc reduced_err{errc::success};
+
+    // For a huge topic with e.g. 100k partitions, this will be a huge loop:
+    // that means we need parallelism, but not so much that we totally
+    // saturate inter-core queues.
+    co_await ss::max_concurrent_for_each(
+      ntps.begin(),
+      ntps.end(),
+      8192,
+      [this, &reduced, &reduced_err](const model::ntp& ntp) -> ss::future<> {
```

Comment:
afaik the consensus is that _everything_ in a coroutine capture list accessed after a co_await in a lambda needs to be copied/moved onto the stack. travis d / ben have a couple good threads on this topic in slack

Reply:
I've turned around three times, sacrificed a goat, and wrapped my reduced state in a lw_shared_ptr struct so that everything is passed by value and I'm not trying to reason about coroutine references.